pkg/podexecutor/staticpod.go

```diff
@@ -317,19 +317,17 @@ func (s *StaticPodConfig) APIServer(_ context.Context, etcdReady <-chan struct{}
 		files = append(files, etcdNameFile(s.DataDir))
 	}
 
-	sockets := []string{}
-	if s.ExternalDatabase {
-		sockets = append(sockets, kineSock(s.DataDir))
-	}
-
 	dirs := onlyExisting(ssldirs)
 	if auditLogFile != "" && auditLogFile != "-" {
 		dirs = append(dirs, filepath.Dir(auditLogFile))
 		excludeFiles = append(excludeFiles, auditLogFile)
 	}
+
+	// Need to mount the entire server directory so that any files recreated in this directory
+	// after the pod has been staretd are not masked by a stale mount
```

Comment:
another nit, sorry 🙃 

```suggestion
	// after the pod has been started are not masked by a stale mount
```